File: backend/crawl/app/models/team_hitter_stats.py
from sqlalchemy import Column, Integer, String, ForeignKey, Float
from sqlalchemy.orm import relationship
from app.models.database import Base


class TeamHitterStats(Base):
    __tablename__ = "team_hitter_stats"

    id = Column(Integer, primary_key=True, index=True)
    rank = Column(Integer)
    year = Column(String(10))
    team_id = Column(Integer, ForeignKey("team.id"))
    avg = Column(String(10))
    game_count = Column(Integer)
    pa = Column(Integer)
    ab = Column(Integer)
    r = Column(Integer)
    h = Column(Integer)
    two = Column(Integer)
    three = Column(Integer)
    homerun = Column(Integer)
    tb = Column(Integer)
    rbi = Column(Integer)
    sac = Column(Integer)
    sf = Column(Integer)
    
    team = relationship("Team", back_populates="team_hitter_stats")

File: backend/baseball/src/main/java/com/newbie/baseball/domain/teamStats/entity/TeamHitterStats.java
package com.newbie.baseball.domain.teamStats.entity;

import com.newbie.baseball.domain.team.entity.Team;
import jakarta.persistence.*;
import lombok.Getter;

@Getter
@Entity
@Table(name = "team_hitter_stats")
public class TeamHitterStats {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    @Column(name = "id")
    private Integer id;

    @Column(name = "`rank`")
    private Integer rank;

    @Column(name = "year", length = 10)
    private String year;

    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "team_id")
    private Team team;

    @Column(name = "avg", length = 10)
    private String avg;

    @Column(name = "game_count")
    private Integer gameCount;

    @Column(name= "pa")
    private Integer pa;

    @Column(name = "ab")
    private Integer ab;

    @Column(name = "r")
    private Integer r;

    @Column(name = "h")
    private Integer h;

    @Column(name = "two")
    private Integer two;

    @Column(name = "three")
    private Integer three;

    @Column(name = "homerun")
    private Integer homerun;

    @Column(name = "tb")
    private Integer tb;

    @Column(name = "rbi")
    private Integer rbi;

    @Column(name = "sac")
    private Integer sac;

    @Column(name = "sf")
    private Integer sf;

}
